app_flutter/test/build_dashboard_page_test.dart

```diff
@@ -54,6 +54,24 @@ void main() {
     buildState.dispose();
   });
 
+  testWidgets('shows settings button', (WidgetTester tester) async {
```

Comment:
This only tests the default filter, it should have a test when given a filter that we expect to change the dashboard.

Reply:
This doesn't even test the filter at all. It just verifies that the icon button is in the tree somewhere so that the property sheet can be brought up.

I'll take this and the previous comment to mean "add some tests that verify that filter properties are affected by the property sheet". It might also make sense to add some tests to test the filter itself independent of display.

Reply:
I added a golden test to verify the look with the default filter. Since the test shows all boxes where text characters might be, I'm not sure it has the ability to show the difference with non-default values.

I also added a whole suite of tests to verify the properties and actions of the `TaskGridFilter` object.

Reply:
I just added another test that builds the TaskGridContainer with a variety of filters and checks the size of the generated grid. This should help determine that the filters are being applied correctly by the widgets. Note that the size of the grid depends on the reproducibility of the testing fake grid generator, but that has to be consistent for a lot of tests to be valid.

Reply:
Sorry, GitHub comment was in a weird place, but looks like you got the message!